### PR TITLE
Add Location feature to Habit Event

### LINKED ISSUE #42: UR-HE #23: User should be able to add an optional location to a habit event upon creation using a map
Expected Behavior:
In the ‘Edit HabitEvent Screen” or “Add HabitEvent Screen”, the user will click the “+” sign to add a location in which a map will be shown to the user to add an optional location.

Mechanism:
The Google Maps API will be used to let the user pick a location for which the habit occurred.

Edge Case Behaviour:
User clicks on the map but doesn’t choose any location -> the current phone’s location will be used as the default location.

Risk: Low
Story Points: 5

User Stories: US 06.01.01, US 06.01.02

### LINKED ISSUE #43: UR-HE #21: User should be able to edit the location of a habit event they own
Risk: Low
Story Points: 3

User Stories: US 02.05.01

### PR DESCRIPTION
**Pull request by Docker-J**
## Description of Changes
Add google map api
Update define habit event view
Add LatLngModel to make firestore stuff easier
Update HabitEventModel

User can use location feature without the permission(just not able to use current location)
If user doesn't pin anything, habit event will be saved without location info.

Location is shown as "City, Province, Country" format in ViewHabitEventActivity

## Checklist

- [ ] Automated tests partial
- [x] Screenshots included (if necessary)
- [x] Code is well commented

Closes #43 
Closes #42

### CODE REVIEW

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/HabitEventModel.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
     /**
      * Location of habit event completion (optional)
      */
-    private String location; // TODO: change to Location type once this is implemented
+    private LatLngModel location; // TODO: change to Location type once this is implemented
```
Comment (j-vent) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/HabitEventModel.java:24]: nit: remove TODO comment

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/LatLngModel.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * Custom Latitude Longitude Class to make HabitEventModelClass serializable
+ */
+public class LatLngModel {
+    private Double latitude;
+    private Double longitude;
+
+    public LatLngModel() {
+
+    }
+
```
Comment (j-vent) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/LatLngModel.java:15]: nit: add javadocs to all methods

File: src/app/src/test/java/com/cmput301f21t09/budgetprojectname/models/WeeklyHabitScheduleModelTest.java [hunk 3]
```diff
@@ -45,19 +45,19 @@ public void testGetAllDays() {
...
-        }
-        Map<String, Object> map = model.toMap();
-        Object innerMap = map.get("weekly");
-        assertNotNull(innerMap);
-    }
+//    /**
+//     * Test for toMap method
+//     */
+//    @Test
+//    public void testToMap() {
```
Comment (j-vent) [inline src/app/src/test/java/com/cmput301f21t09/budgetprojectname/models/WeeklyHabitScheduleModelTest.java:52]: not sure why this test is commented out

**Review by j-vent — DISMISSED** [review]
Great work!! Just some small comments + I'll reapprove

**Review by cheetiantey — DISMISSED** [review]
Looks great overall!!

File: src/app/src/test/java/com/cmput301f21t09/budgetprojectname/models/WeeklyHabitScheduleModelTest.java (hunk 3 above)
Comment (Docker-J) [reply src/app/src/test/java/com/cmput301f21t09/budgetprojectname/models/WeeklyHabitScheduleModelTest.java:52]: Mac changes testToMap to testToList so this test not working anymore. I think I commented out when I was updating other test as this keep throw the error. I reverted it

**Review by cheetiantey — APPROVED** [review]
LGTM!

**Review by loravocado — APPROVED** [review]
Looks so good!!

**Pull request merged**